spec/lib/wos_records_spec.rb

```diff
@@ -0,0 +1,199 @@
+      expect(result.first).to be_an String
+    end
+    it 'works with decoded records' do
+      result = wos_records_decoded.uids
+      expect(result).to be_an Array
+      expect(result.first).to be_an String
+    end
+  end
+
+  describe '#duplicate_uids' do
+    let(:dup_uids) { wos_recordsA.duplicate_uids(wos_recordsB) }
+
+    it 'works' do
+      expect(dup_uids).to be_an Set
+    end
+    it 'returns duplicate UIDs' do
+      expect(dup_uids.to_a).to eq ['WOS:A2']
+    end
+    it 'does not modify records' do
+      expect { dup_uids }.not_to change { wos_recordsA.uids }
+    end
+    it 'does not modify input records' do
+      expect { dup_uids }.not_to change { wos_recordsB.uids }
+    end
+  end
+
+  describe '#duplicate_records' do
+    let(:dup_records) { wos_recordsA.duplicate_records(wos_recordsB) }
+
+    it 'works' do
+      expect(dup_records).to be_an Nokogiri::XML::NodeSet
+    end
+    it 'returns duplicate record nodes' do
+      uid = dup_records.search('UID').text
+      expect(uid).to eq 'WOS:A2'
+    end
+    it 'does not modify records' do
+      expect { dup_records }.not_to change { wos_recordsA.uids }
+    end
+    it 'does not modify input records' do
+      expect { dup_records }.not_to change { wos_recordsB.uids }
+    end
+  end
+
+  describe '#new_records' do
+    let(:new_records) { wos_recordsA.new_records(wos_recordsB) }
+
+    it 'works' do
+      expect(new_records).to be_an Nokogiri::XML::NodeSet
+    end
+    it 'returns new record nodes' do
+      uid = new_records.search('UID').text
+      expect(uid).to eq 'WOS:B2'
+    end
+    it 'does not modify records' do
+      expect { new_records }.not_to change { wos_recordsA.uids }
+    end
+    it 'does not modify input records' do
+      expect { new_records }.not_to change { wos_recordsB.uids }
+    end
+  end
+
+  describe '#merge_records' do
+    let(:merge_records) { wos_recordsA.merge_records(wos_recordsB) }
+
+    it 'works' do
+      expect(merge_records).to be_an described_class
+    end
+    it 'has merged records' do
+      expect(merge_records.uids).to eq %w(WOS:A1 WOS:A2 WOS:B2)
+    end
+    # Immutability specs
+    it 'does not modify records' do
+      expect { merge_records }.not_to change { wos_recordsA.uids }
+    end
+    it 'does not modify input records' do
+      expect { merge_records }.not_to change { wos_recordsB.uids }
+    end
+  end
+
+  describe '#print' do
+    it 'works with encoded records' do
+      expect { wos_records_encoded.print }.to output.to_stdout
+    end
+    it 'works with decoded records' do
+      expect { wos_records_decoded.print }.to output.to_stdout
+    end
+  end
+
+  describe '#rec_nodes' do
+    let(:rec_nodes) { wos_records_decoded.rec_nodes }
+
+    it 'works' do
+      expect(rec_nodes).not_to be_nil
+    end
+  end
+
+  describe '#to_xml' do
+    it 'works with encoded records' do
+      result = wos_records_encoded.to_xml
+      expect(result).to be_an String
```

Comment:
Can we get a stronger expectation here than just "gimme a String"?  Since we are dealing with encoding, it is sorta important that not all strings are equivalent.

Reply:
Maybe an XSD validation for the record data?

Reply:
Or at least expect it not to contain `&lt;` or something.

Reply:
I don't think full XSD is required.  Just look for something that you expect the encoding to adjust (or not adjust) and target it.